### PR TITLE
IController changes syncup 

### PR DESCRIPTION
**Pull request by HaseenaSainul**

### CODE REVIEW

File: Source/WPEFramework/Controller.cpp [hunk 1]
```diff
@@ -183,6 +183,132 @@ namespace Plugin {
...
+        // Normalise return code
+        if (result != Core::ERROR_NONE) {
+            result = Core::ERROR_GENERAL;
+        }
+
+        return result;
+
+    }
+
+    uint32_t Controller::Delete(const string& path)
```
Comment (MFransen69) [inline Source/WPEFramework/Controller.cpp:213]: I see that the delete code is now different then it was previously in endpoint_delete. For all the moved code let's just keep the implementation that was already in there (didnt check the others that were move3d, so please check these as well)

File: Source/WPEFramework/PluginServer.cpp [hunk 2]
```diff
@@ -250,7 +250,10 @@ namespace PluginHost
 
         _adminLock.Unlock();
 
-        ASSERT(_notifiers.size() == 0);
+        TRACE_L1("Pending notifiers are %lu", _notifiers.size());
```
Comment (MFransen69) [inline Source/WPEFramework/PluginServer.cpp]: Let's also not add this code as it was not in R2 yet. I only added the unused variable in R3

File: Source/core/FileSystem.cpp [hunk 3]
```diff
@@ -56,6 +56,99 @@ namespace Core {
     {
         Close();
     }
+
```
Comment (MFransen69) [inline Source/core/FileSystem.cpp:59]: we should not add these changes to R2

File: Source/core/FileSystem.h [hunk 4]
```diff
@@ -156,6 +156,7 @@ namespace Core {
             return (*this);
         }
 
+        static string Normalize(const string& input, bool& valid);
```
Comment (MFransen69) [inline Source/core/FileSystem.h]: Let's also not add this

**Review by MFransen69 — CHANGES_REQUESTED** [review]

File: Source/WPEFramework/PluginServer.cpp (hunk 2 above)
Comment (HaseenaSainul) [reply Source/WPEFramework/PluginServer.cpp]: It is just a cleanup, hence added. will remove

File: Source/WPEFramework/Controller.cpp (hunk 1 above)
Comment (HaseenaSainul) [reply Source/WPEFramework/Controller.cpp:213]: will check and update

File: Source/core/FileSystem.cpp (hunk 3 above)
Comment (HaseenaSainul) [reply Source/core/FileSystem.cpp:59]: removed

File: Source/core/FileSystem.h (hunk 4 above)
Comment (HaseenaSainul) [reply Source/core/FileSystem.h]: removed

File: Source/WPEFramework/Controller.cpp (hunk 1 above)
Comment (HaseenaSainul) [reply Source/WPEFramework/Controller.cpp:213]: The update required for only Delete method, I removed Normalise call usage instead used older implementation. In Clone() method, the changes I added is just renaming variable to camel case

File: Source/core/FileSystem.cpp (hunk 3 above)
Comment (HaseenaSainul) [reply Source/core/FileSystem.cpp:59]: removed

File: Source/WPEFramework/PluginServer.cpp (hunk 2 above)
Comment (HaseenaSainul) [reply Source/WPEFramework/PluginServer.cpp]: updated

**Review by MFransen69 — APPROVED** [review]

**Review by karuna2git — APPROVED** [review]

**Pull request merged**